src/script/src/python/coprocessor/compile.rs

```diff
@@ -31,23 +31,40 @@ fn create_located<T>(node: T, loc: Location) -> Located<T> {
 /// generate a call to the coprocessor function
 /// with arguments given in decorator's `args` list
 /// also set in location in source code to `loc`
-fn gen_call(name: &str, deco_args: &DecoratorArgs, loc: &Location) -> ast::Stmt<()> {
+fn gen_call(
+    name: &str,
+    deco_args: &DecoratorArgs,
+    kwarg: &Option<String>,
+    loc: &Location,
+) -> ast::Stmt<()> {
     let mut loc = loc.to_owned();
     // adding a line to avoid confusing if any error occurs when calling the function
     // then the pretty print will point to the last line in code
     // instead of point to any of existing code written by user.
     loc.newline();
-    let args: Vec<Located<ast::ExprKind>> = deco_args
-        .arg_names
-        .iter()
-        .map(|v| {
-            let node = ast::ExprKind::Name {
-                id: v.to_owned(),
-                ctx: ast::ExprContext::Load,
-            };
-            create_located(node, loc)
-        })
-        .collect();
+    let mut args: Vec<Located<ast::ExprKind>> = if let Some(arg_names) = &deco_args.arg_names {
+        arg_names
+            .iter()
+            .map(|v| {
+                let node = ast::ExprKind::Name {
+                    id: v.to_owned(),
+                    ctx: ast::ExprContext::Load,
+                };
+                create_located(node, loc)
+            })
+            .collect()
+    } else {
+        vec![]
+    };
+
+    if let Some(kwarg) = kwarg {
+        let node = ast::ExprKind::Name {
+            id: kwarg.to_owned(),
```

Comment:
Could we just use clone
```suggestion
                    id: v.clone(),
                    ctx: ast::ExprContext::Load,
                };
                create_located(node, loc)
            })
            .collect()
    } else {
        vec![]
    };

    if let Some(kwarg) = kwarg {
        let node = ast::ExprKind::Name {
            id: kwarg.clone(),
```